Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -19,12 +20,16 @@
     /// </summary>
     public class RequestTrackingTelemetryModule : ITelemetryModule
     {
-        private readonly IList<string> handlersToFilter = new List<string>();
+        private const string CorrelationStatusHeartbeatProperty = "CorrelationHealthStatus_RequestWithContent";
+
+        // if HttpApplicaiton.OnRequestExecute is available, we don't attempt to detect any correlation issues
+        private static bool correlationIssuesDetectionComplete = typeof(HttpApplication).GetMethod("OnExecuteRequestStep") != null;
```

Comment:
Is there an exact version of .net framework when this will be available? I believe heartbeat already sents out .NET Framework information - can we use that alone ?

Reply:
As we have seen in func tests, .NET version does not really matter. What matters is the version of ASP.NET and even then, customers that do not have POST, PUT or PATCH requests (e.g. serving static websites) would never experience the issue. 
We can guess that there are not much of such customers, but I'd prefer to measure it rather than guessing.